integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala

```diff
@@ -42,15 +42,46 @@ class CarbonSession(@transient val sc: SparkContext,
     this(sc, None)
   }
 
+
+  // SessionStateCodeGenerateFactory.init(sc.version)
+  // CarbonOptimizerCodeGenerateFactory.init(sc.version)
+  // val carbonDefaultOptimizer = CarbonOptimizerCodeGenerateFactory.getInstance()
+  //  .carbonoptimizerFactory.createCarbonOptimizer()
+  //  @transient
+  //  override lazy val sessionState: SessionState = new CarbonSessionState(this)
+
+
+
+  def getSessionState(sparkContext: SparkContext): SessionState = {
+    if (sparkContext.version.contains("2.1")) {
```

Comment:
It'd better use sparkContext.version.**startsWith**("2.1"), if version = 2.2.1, contains("2.1") will return  true.